src/app/api/entry/[_id]/roles/route.ts

```diff
@@ -0,0 +1,34 @@
+import { NextRequest, NextResponse } from "next/server";
+import connectDB from "@database/db";
+import VolunteerRoles from "@database/volunteerRoleSchema";
+import entrySchema from "@database/volunteerEntrySchema";
+
+type IParams = {
+  params: {
+    _id: string;
+  };
+};
+
+// GET all VolunteerRoles for a specific volunteer entry
+export async function GET(req: NextRequest, { params }: IParams) {
+  await connectDB();  // function from db.ts
+  const { _id } = params;
+
+  try {
+    const entry = await entrySchema.findOne({ _id: _id }).orFail();
+    const volunteerRolesID = entry.roles;
+
+    const volunteerRoles = [];
+    for (let i = 0; i < volunteerRolesID.length; i++) {   // finding each role
+      const role = await VolunteerRoles.findById(volunteerRolesID[i]);
+      if (role) {
+        volunteerRoles.push(role);
+      }
+    }
```

Comment:
Some points of optimization  you might find helpful but not necessary for this PR to be approved:
- volunteerRolesID is a confusing variable name for an array. It should be changed to something like "volunteerRolesIDArr",...
- I just did some searching and it seems like you can do a single request instead of multiple requests in a for loop:
`const volunteerRoles = await VolunteerRoles.find({ _id: { $in: volunteerRolesID } });`
`$in` lets you pass in an array of ids you want to find matches of.